plugins/woocommerce/src/Packages.php

```diff
@@ -47,20 +47,41 @@ private function __construct() {}
 		'woocommerce-gutenberg-products-block' => '\\Automattic\\WooCommerce\\Blocks\\Package',
 	);
 
+	/**
+	 * Array of plugin names and their main plugin classes.
+	 *
+	 * Once a plugin has been merged into WooCommerce Core it should have its slug added here. This will ensure
+	 * that we deactivate the feature plugin automatically to prevent any problems caused by conflicts between
+	 * the two versions both being active.
+	 *
+	 * @var array Key is the package name/directory, value is the main plugin class which handles init.
+	 */
+	protected static $merged_plugins = array(
+		'woocommerce-brands' => '\\Automattic\\WooCommerce\\Internal\\Brands',
```

Comment:
How would you feel about changing the name of `$merged_packages` instead? The point of the feature in the first place was because the _plugin_ was merged after all!

Reply:
Hey @ObliviousHarmony! 

The main reason I went with a different array/name here is that the `initialize_packages` [function](https://github.com/woocommerce/woocommerce/blob/c11a14629cacc7b32a2882520014c6eda2a12398/plugins/woocommerce/src/Packages.php#L205-L235) seems to be doing a lot more than [what I need](https://github.com/woocommerce/woocommerce/blob/c11a14629cacc7b32a2882520014c6eda2a12398/plugins/woocommerce/src/Packages.php#L242-L246) for Brands. Do you think it is OK to let `initialize_packages` to run for my own, "dummy" package or do you have any other suggestions on how to run only part of `initialize_packages` for Brands?

Reply:
I also need plugins, like Brands, in a separate array to run functions like these: 
- https://github.com/woocommerce/woocommerce/pull/50165/files#diff-c3a93b21b4c564d9c866729f93fcaf38838c2a67ab30e5cf8a0bd4db7e269fdcR147
- https://github.com/woocommerce/woocommerce/pull/50165/files#diff-c3a93b21b4c564d9c866729f93fcaf38838c2a67ab30e5cf8a0bd4db7e269fdcR169
- https://github.com/woocommerce/woocommerce/pull/50165/files#diff-c3a93b21b4c564d9c866729f93fcaf38838c2a67ab30e5cf8a0bd4db7e269fdcR188

Reply:
Hey @ObliviousHarmony, @jimjasson is AFK this week, so I'm hopping on here.

As a fresh pair of eyes, I think it makes a lot of sense to change `$merged_packages` to `$merged_plugins` and use the UX introduced in this PR.  

I see that `$merged_plugins` is also calling the `init()` method on the plugin's main class, so it does seem like duplicate code.

I'll take a stab at cleaning it up and de-dupping things.

Reply:
Changes done; please review them :) See https://github.com/woocommerce/woocommerce/pull/50165/files/b320687c68023aa82b045c5de52adcedb93d2160..ff07d4c566951fbde0f7265ec099a58c1bd6a4da

Reply:
Thanks @Dakota, that looks great!